roles/create-ai-environment/tasks/main.yaml

```diff
@@ -0,0 +1,289 @@
+
+- name: Generate new output folder if non-exist
+  when: output_folder_name is not defined
+  file:
+    path: '{{ ansible_user_dir }}/{{ zuul.project.src_dir }}/{{ user_file }}/training_result'
+    state: directory
+
+- name: Set training output folder name by generation
+  when: output_folder_name is not defined
+  set_fact:
+    training_output_folder_name: "training_result"
+
+- name: Install {{ runtime.stdout }} Runtime
+  when: runtime_name
+  shell:
+    cmd: |
+      set -x
+      set -e
+      runtime_name={{ runtime_name.stdout }}
+      runtime_version={{ runtime_version.stdout }}
+      add-apt-repository -y ppa:deadsnakes/ppa
+      apt-key adv --keyserver keyserver.ubuntu.com --recv-keys F60F4B3D7FA2AF80
+      apt update
+      if [ $runtime_name == 'python' ];then
+        apt install -y $runtime_name$runtime_version
+        apt install -y python2.7 python3.5 python3.6 python3.7
+      fi
+
+      if [ $runtime_name == 'java' ];then
+        apt install -y default-jre default-jdk
+        apt install -y openjdk-7-jre openjdk-8-jdk
+      fi
+
+      if [ $runtime_name == 'c' ];then
+        apt install -y gcc=$runtime_version g++=$runtime_version
+        apt install -y gcc-7 g++-7 gcc-8 g++-8 gcc-9 g++-9
+      fi
+
+      if [ $runtime_name == 'go' ];then
+        apt install -y golang-go=$runtime_version
+      fi
+
+    executable: /bin/bash
+
+- name: Prepare GPU env (GPU driver and CUDA)
+  when: is_gpu_enabled
+  shell:
+    cmd: |
+      set -xe
+      apt-key adv --fetch-keys http://developer.download.nvidia.com/compute/cuda/repos/ubuntu1604/x86_64/7fa2af80.pub
+      wget http://developer.download.nvidia.com/compute/cuda/repos/ubuntu1604/x86_64/cuda-repo-ubuntu1604_9.1.85-1_amd64.deb
+      cd /var/
+      apt purge -y cuda*
+      apt-get remove -y --purge nvidia-418 nvidia-modprobe nvidia-settings
+      cd -
+      apt install -y ./cuda-repo-ubuntu1604_9.1.85-1_amd64.deb
+      wget http://developer.download.nvidia.com/compute/machine-learning/repos/ubuntu1604/x86_64/nvidia-machine-learning-repo-ubuntu1604_1.0.0-1_amd64.deb
+      apt install -y ./nvidia-machine-learning-repo-ubuntu1604_1.0.0-1_amd64.deb
+      apt update
+      apt install -y cuda9.0 cuda-cublas-9-0 cuda-cufft-9-0 cuda-curand-9-0 cuda-cusolver-9-0 cuda-cusparse-9-0 libcudnn7=7.2.1.38-1+cuda9.0 libnccl2=2.2.13-1+cuda9.0 cuda-command-line-tools-9-0
+      apt install -y linux-headers-4.4.0-142-generic
+      wget http://us.download.nvidia.com/tesla/384.81/NVIDIA-Linux-x86_64-384.81.run
+      chmod +x ./NVIDIA-Linux-x86_64-384.81.run
+      cat > 'input' <<EOF
+      yes
+      yes
+      EOF
+      ./NVIDIA-Linux-x86_64-384.81.run -a --ui=none < ./input
+    executable: /bin/bash
+
+- name: Install {{ framework.stdout }}-{{ framework_version.stdout }}
+  when:
+    ( framework.stdout == "tensorflow" and runtime_name.stdout in ["python", "c", "java", "go"] ) or
+    ( framework.stdout == "pytorch" and runtime_name.stdout in ["python", "c"] )
```

Comment:
Will be good to refactor this two conditions into a variable? then you can use them later and do not have write so many words every time?

Reply:
yes, kiwik already mentioned that.